### ISSUE TITLE
Pymt_Sedflux

### ISSUE THREAD

**Issue by XianaZhang** [csdms/help-desk#22]
Hi there, 
      Could i ask a help for my last questions. Why the maximum grain size for bed load the the sedflux model allows is only 0.005m which seems unreasonable？
       
bed_load_sediment_grain_size:
  description: Grain size of bed-load sediment
  value:
    default:  400.0
    range:
      max:  5000.0
      min:  0.1
    type:  float
    units:  um

     Should we make the compaction process file stay open because there is a compaction cofficient in sediment file?
      Does it possible to use sedflux2d under pymt environment? I did not find a manule about how to use it.

**Comment by mcflugen** [comment]
@XianaZhang There is no reason the maximum value for the grain size is 5000 microns. It's arbitrary and should probably be adjusted to be larger.

Even though there is a compaction coefficient in the sediment file, it doesn't mean the compaction process should be turned on. If it's off, the coefficient is ignored.

Unfortunately, *sedflux2d* is not available through *pymt*.

**Comment by XianaZhang** [comment]
@mcflugen Thank you very much! I wonder if there is a upper limit for the maximum and minimum value of the bed-load grain size? Because i saw there is a file named parameters.yaml in the template input files for sedflux 3d as seen below, there is a maximum and minimum value defined and i just always thought that means a range for the bed-load grain size. So what's the real meaning of them?
 
   bed_load_sediment_grain_size:
   description: Grain size of bed-load sediment
    value:
    default: 400.0
    range:
      max: 5000.0
      min: 0.1
     type: float
     units: um
    
     I also want to know is there an upper limit for the number of sediment grain types, seasons for river and epoches? 
      And there are some other input variables which are not  mentioned in sedflux documentation, for example the variable 'channel_exit_x-section_mean_of_depth'. Because there are also the variable 'depth'  in river file, i just don't know what's the relationship of them. If the delta morphology is more controled by the 'channel_exit_x-section_mean_of_depth' than 'depth' which is defined in river file ? And also what's the maximum and minimum value for these 'other input variables'?

**Comment by XianaZhang** [comment]
@mcflugen Except the questions above, i also have my last two questions. I feel a little confused by the plum process file. I saw there is a description for it in the sedflux documents as seen below. Does that mean if i want to run plum model alone, i need to download the two files? And if i just run the sedflux3d model, Does i also need to download them ?


_**To run the plume model you will need two input files. I’ve create an example of each to get you started:
plume_config.kvf: describes some physical and numerical constants used by the model.
plume_flood.kvf: describes input conditions to the model._**


And there is a function named model.parameters. I wonder what it works for? Because it seems that it's output is the variables and their values of the template input files, not the new input files which  we created by setup functions and have changed the values and variables for them by manule.

**Comment by mcflugen** [comment]
@XianaZhang Yes, if you would like to run the *plume* model alone, you will need to download the additional files. These files provide river and sediment information to *plume*. If you run *sedflux*, this information is provided by *sedflux*.

The *model.parameters* attribute provides a list of the available parameters that you can set, along with their defaults. It doesn't provide the new input values.

**Comment by mcflugen** [comment]
@XianaZhang Currently, the upper limit for *bed_load_sediment_grain_size* is 5000 um and the minimum is 0.1. As I said, there is no particular reason for this upper limit and it should probably be increased as it's probably too small.

You can increase the number of grain types to as many as you like, however, I've never used any more than 5. There also aren't any limits to the number of seasons or epochs (at least not that I can recall).

The *channel_exit_x-section_mean_of_depth* variable is the name used for setting the river depth programmatically through the model's *set_value* method. The *depth* in the river file has the same meaning but is set during the model setup and so is static. There are no strict maximum and minimum values for these "other input values", other than they should just be typical values.

**Comment by XianaZhang** [comment]
@mcflugen Okay! Thank you! Does that mean in fact there is no strict maximum and minimum values for bed-load grain size or for other varibles? For example, sedflux will allowe me to set 5000000 um for bed-load size if i wants?

**Comment by mcflugen** [comment]
@XianaZhang Yes, I believe so. *sedflux* shouldn't have a problem with a larger grain size, at least not that I can remember. The max and min are only when setting the grain size through the *Sedflux3D* *setup* method.

**Comment by XianaZhang** [comment]
Oh！Thank you!  So if i want to set a larger grain size, i am just allowed to set it by manule for my new created input files?  It will not work if i use setup method.

**Comment by XianaZhang** [comment]
@mcflugen  Oh！Thank you! So if i want to set a larger grain size, am i just allowed to set it by manule for my new created input files? It will not work if i use setup method.

**Comment by XianaZhang** [comment]
@mcflugen My final question is you said channel_exit_x-section_mean_of_depth variable and all these "other input values" is the same as the variables in river input file. The difference between them is just they have different set methods. So does that mean in fact there is no need for me to set values for  channel_exit_x-section_mean_of_depth variable and all these "other input values", because i can just choose to set values in rivers input file? If i set values for these "other input values", will it overlay the values i set in river input file?
